lib/compress.js

```diff
@@ -4858,6 +4858,12 @@ merge(Compressor.prototype, {
             }
             break;
         }
+        // void 0 == x => null == x
+        if (compressor.option("comparisons") &&
+            (self.operator == "==" || self.operator == "!=") &&
+            is_undefined(self.left, compressor)) {
+            self.left = make_node_from_constant(null, self.left);
```

Comment:
Please move this code to below:

https://github.com/mishoo/UglifyJS2/blob/b16380d66961f408932de781a3425d2992ec51b3/lib/compress.js#L4766-L4767

so the first two conditions are not needed.

Reply:
Please use `make_node(AST_Null, self.left)` as it is more efficient.

Reply:
@kzc I can't move it up, because the line before that is `// XXX: intentionally falling down to the next case`.

I'll change it to `make_node(AST_Null, self.left)` shortly.

Reply:
> I can't move it up, because the line before that is // XXX: intentionally falling down to the next case.

It would be inefficient to repeat all those string comparisons again. You can set a flag in the `falling down` case and check it is not set in the `"=="` and `"!="` cases.

Reply:
Alright, I'll do that.